packages/p2p/src/components/my-profile/payment-methods/add-payment-method/add-payment-method-form.jsx

```diff
@@ -2,26 +2,43 @@ import React from 'react';
 import classNames from 'classnames';
 import PropTypes from 'prop-types';
 import { observer } from 'mobx-react-lite';
-import { Field, Form, Formik } from 'formik';
+import { Field, Form, Formik, useFormikContext } from 'formik';
 import { Button, Icon, Input, Loading, Modal, Text } from '@deriv/components';
 import { usePaymentMethodValidator } from 'Components/hooks';
 import { Localize, localize } from 'Components/i18next';
 import { useStores } from 'Stores';
 
-const AddPaymentMethodForm = ({ formik_ref, should_show_separated_footer = false }) => {
+const RestoreFormHistory = () => {
+    const { setValues } = useFormikContext();
+    const { my_profile_store } = useStores();
+
+    React.useEffect(() => {
+        if (my_profile_store.formik_history) {
+            // we have to explicitly call setValues if there is form history already saved
+            // passing in the form history to Formik's initialValues will not suffice since Formik will not treat it as 'dirty'
+            // thus this will cause bugs such as Add button being disabled even though user has already typed something previously
+            setValues(my_profile_store.formik_history.values);
+        }
+    }, []);
+    return null;
+};
+
+const AddPaymentMethodForm = ({ should_show_separated_footer = false }) => {
     const { my_ads_store, my_profile_store } = useStores();
     const validateFields = usePaymentMethodValidator();
 
+    const formikRef = node => {
+        if (node !== null) {
```

Comment:
explicitly update the formik's ref when the ref changes, or else changes occuring within Formik and the ref passed from `innerRef` prop won't be reflected with the observable `formik_ref`

Reply:
@adrienne-deriv please check if it's gonna work for you🙂
```suggestion
        if (node) {
```